pytools/cache.py

```diff
@@ -0,0 +1,273 @@
+MAX_KEY_LENGTH = 250
+CONTENT_ENCODING = "utf-8"
+
+
+def init(servers, namespace):
+    """Initializes memcached client. Needs to be called before use.
+
+    Args:
+        servers: List of strings with memcached server addresses (host:port).
+        namespace: Global namespace that will be prepended to all keys.
+    """
+    global _mc, _glob_namespace
+    _mc = HashClient(
+        servers=servers,
+        serializer=_serializer,
+        deserializer=_deserializer,
+    )
+    _glob_namespace = namespace + ":"
+    _glob_namespace = _glob_namespace.encode("ascii")
+    if len(_glob_namespace) + SHA1_LENGTH > MAX_KEY_LENGTH:
+        raise ValueError("Namespace is too long.")
+
+
+def set(key, val, time=0, namespace=None):
+    """Set a key to a given value.
+
+    Args:
+        key: Key of the item.
+        val: Item's value.
+        time: The time after which this value should expire, either as a delta
+            number of seconds, or an absolute unix time-since-the-epoch value.
+            If set to 0, value will be stored "forever".
+        namespace: Optional namespace in which key needs to be defined.
+
+    Returns:
+        True if stored successfully.
+    """
+    if _mc is None: return
+    return set_many({key: val}, time, namespace)
+
+
+def get(key, namespace=None):
+    """Retrieve an item.
+
+    Args:
+        key: Key of the item that needs to be retrieved.
+        namespace: Optional namespace in which key was defined.
+
+    Returns:
+        Stored value or None if it's not found.
+    """
+    if _mc is None: return
+    result = get_many([key], namespace)
+    return result.get(key)
+
+
+def delete(key, namespace=None):
+    """Delete an item.
+
+    Args:
+        key: Key of the item that needs to be deleted.
+        namespace: Optional namespace in which key was defined.
+
+    Returns:
+          True if deleted successfully.
+    """
+    if _mc is None: return
+    return delete_many([key], namespace) == 1
+
+
+def set_many(mapping, time=0, namespace=None):
+    """Set multiple keys doing just one query.
+
+    Args:
+        mapping: A dict of key/value pairs to set.
+        time: Time to store the keys (in milliseconds).
+        namespace: Namespace for the keys.
+
+    Returns:
+        True on success.
+    """
+    if _mc is None: return
+    return _mc.set_many(_prep_dict(mapping, namespace), time)
+
+
+def get_many(keys, namespace=None):
+    """Retrieve multiple keys doing just one query.
+
+    Args:
+        keys: Array of keys that need to be retrieved.
+        namespace: Namespace for the keys.
+
+    Returns:
+        A dictionary of key/value pairs that were available.
+    """
+    if _mc is None: return {}
+    res = _mc.get_many(_prep_list(keys, namespace))
+    for k in keys:
+        if namespace:
+            namespace_and_version = _append_namespace_version(namespace)
+        else:
+            namespace_and_version = None
+        prep = _prep_key(k, namespace_and_version)
```

Comment:
That's all what `_prep_list` did above right? Would it make sense to just zip `keys` with `_prep_list(keys, namespace)`?